launch/test/launch/test_emulate_tty.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for emulate_tty configuration of ExecuteProcess actions."""
+
+import sys
+
+import launch
+import pytest
+
+
+class OnExit(object):
+
+    def __init__(self):
+        self.returncode = None
+
+    def handle(self, event, context):
+        self.returncode = event.returncode
+
+
+@pytest.mark.parametrize('test_input,expected', [
+    (None, 1),  # use the default defined by ExecuteProcess
+    ('true', 1),  # redundantly override the default via LaunchConfiguration
+    ('false', 0)  # override the default via LaunchConfiguration
+])
+def test_emulate_tty(test_input, expected):
+    on_exit = OnExit()
+    ld = launch.LaunchDescription()
+    ld.add_action(launch.actions.ExecuteProcess(
+        cmd=[sys.executable, '-c', 'import sys; sys.exit(sys.stdout.isatty())'])
+    )
+    if test_input is not None:
+        ld.add_action(launch.actions.SetLaunchConfiguration('emulate_tty', test_input))
+    ld.add_action(
+        launch.actions.RegisterEventHandler(
+            launch.event_handlers.OnProcessExit(on_exit=on_exit.handle)
+        )
+    )
+    ls = launch.LaunchService()
+    ls.include_launch_description(ld)
+    ls.run()
+    assert on_exit.returncode == expected
```

Comment:
This is actually what is failing. The other output is a problem, but not the reason for the failure. See:

```
test\launch\test_emulate_tty.py:53: in test_emulate_tty
    assert on_exit.returncode == expected
E   assert 0 == 1
E    +  where 0 = <test_emulate_tty.OnExit object at 0x000002340EE160B8>.returncode
```

Basically `isatty()` is returning false on Windows (kind of expected) for one or more of your parameterized cases.

Reply:
Aye, thanks for posting the details. I got to the same conclusions after doing some more reading last night.

How would you like to handle it? Options:

1. Dig deep, get tty emulation actually working and with unicode on windows (winpty?)
2. Refuse to attempt tty emulation (with warning?) on windows, adjust test expectations for windows with comments, raise an issue
3. Adjust test expectations for windows with comments (but perhaps there are risks with having gone down the fork of acting on tty emulation which cause problems), raise an issue
4. Other?

I think I'd lean towards 2. to avoid introducing unknown problems and because I don't have the resources to chase down the windows problems.

Reply:
Number 2 or 3 is fine by me. The underlying API already will silently degrade when `import pty` fails, as it should on Windows.